File: src/js/mqtt-tls-subscriber.js
var mqtt = require('mqtt');
var fs = require('fs');
var KEY = '../../certs/client.key';
var CERT = '../../certs/client.crt';
var TRUSTED_CA_LIST = '../../certs/ca.crt';
var HOST = '192.168.1.1';
var PORT = 8883;

var options = {
  host: HOST,
  port: PORT,
  keyPath: KEY,
  certPath: CERT,
  rejectUnauthorized : false,
  ca: TRUSTED_CA_LIST
};

var client = mqtt.createSecureClient(options);

client.subscribe('sensors/temperature/data');

client.on('connect', function(){
	console.log("Connected to MQTT broker at " + HOST + ":" + PORT);
});

client.on('message', function(topic, message) {
  console.log("Recevied: " + message.toString());
});

File: src/python/mqtt-tls-subscriber.py
import sys
import paho.mqtt.client as mqtt

def on_connect(client, obj, flags, rc):
    print("Connected to MQTT broker at: " + str(rc))

def on_message(client, obj, msg):
    print("Connected to MQTT broker at " + str(msg.payload))

def on_publish(client, obj, mid):
    print("Pubished: " + str(mid))

def on_subscribe(client, obj, mid, granted_qos):
    print("Subscribed: sensors/temperature/data")

def on_log(client, obj, level, string):
    print(string)

client = mqtt.Client()
client.on_message = on_message
client.on_connect = on_connect
client.on_publish = on_publish
client.on_subscribe = on_subscribe

client.tls_set('../../certs/ca.crt',
    '../../certs/client.crt',
    '../../certs/client.key')

client.connect("192.168.1.1", 8883, 60)
client.subscribe("sensors/temperature/data", 0)

client.loop_forever()
